Src/AmrCore/AMReX_Interp_3D_C.H

```diff
@@ -617,6 +617,60 @@ void ccprotect_3d (int ic, int jc, int kc, int nvar,
 
 }
 
+template <typename T>
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+void ccquartic_interp (int i, int j, int k, int n,
+                       IntVect const& /* ratio */,
+                       Array4<T const> const& crse,
+                       Array4<T>       const& fine) noexcept
+{
+    // Note: there are asserts in CellConservativeQuartic::interp()
+    //       to check whether ratio is all equal to 2.
+
+    const Array1D<Real, -2, 2> cL = { -0.01171875,  0.0859375, 0.5, -0.0859375, 0.01171875 };
+
+    int ic = amrex::coarsen(i,2);
+    int jc = amrex::coarsen(j,2);
+    int kc = amrex::coarsen(k,2);
+    int irx = std::abs(i % 2);
+    int jry = std::abs(j % 2);
+    int krz = std::abs(k % 2);
+
+    Array1D<Real,               0, 1> ftmp;
+    Array2D<Real,        -2, 2, 0, 1> ctmp;
+    Array3D<Real, -2, 2, -2, 2, 0, 1> ctmp2;
```

Comment:
Like in 1d and 2d, we can reduce the dimension of these by 1.  This will use less memory.